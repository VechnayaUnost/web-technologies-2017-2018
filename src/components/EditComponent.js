import React from 'react';
import { connect } from "react-redux";
import { Tab, Tabs, TabList, TabPanel } from 'react-tabs';

class EditComponent extends React.Component {
    constructor() {
        super();
        this.state = { tabIndex: 0 };
    }

    render() {
        return (
            <Tabs selectedIndex={this.state.tabIndex} onSelect={tabIndex => this.setState({ tabIndex })}>
                <TabList>
                    <Tab>Основное</Tab>
                    <Tab>Образование</Tab>
                    <Tab>Контакты</Tab>
                </TabList>
                <TabPanel><button >Edit1</button></TabPanel>
                <TabPanel><button >Edit2</button></TabPanel>
                <TabPanel><button >Edit3</button></TabPanel>
                <textarea/>
            </Tabs>
        );
    }
}

export default connect(null)(EditComponent);