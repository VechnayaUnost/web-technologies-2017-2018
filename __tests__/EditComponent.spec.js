import React from 'react';
import EditComponent from '../src/components/EditComponent';
import { shallow, mount, configure } from 'enzyme';
import { shallowToJson, mountToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('EditComponent component test', () => {
    it('EditComponent component rendered succesfully', () => {
        const editComponent = mount(
            <EditComponent />
        );
        expect(mountToJson(editComponent)).toMatchSnapshot();
    });

    it('EditComponent component tabs switch', () => {
        const editComponent = shallow(
            <EditComponent />
        );
        editComponent.find('Tabs').simulate('select', 1);
        expect(editComponent.props().selectedIndex).toBe(1);
    });
});