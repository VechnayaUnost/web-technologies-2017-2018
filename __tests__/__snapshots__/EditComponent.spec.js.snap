// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditComponent component test EditComponent component rendered succesfully 1`] = `
<EditComponent>
  <Tabs
    defaultFocus={false}
    defaultIndex={null}
    forceRenderTabPanel={false}
    onSelect={[Function]}
    selectedIndex={0}
  >
    <UncontrolledTabs
      className="react-tabs"
      focus={false}
      forceRenderTabPanel={false}
      onSelect={[Function]}
      selectedIndex={0}
    >
      <div
        className="react-tabs"
        data-tabs={true}
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        <TabList
          className="react-tabs__tab-list"
          key=".0"
        >
          <ul
            className="react-tabs__tab-list"
            role="tablist"
          >
            <Tab
              className="react-tabs__tab"
              disabledClassName="react-tabs__tab--disabled"
              focus={false}
              id="react-tabs-0"
              key=".0"
              panelId="react-tabs-1"
              selected={true}
              selectedClassName="react-tabs__tab--selected"
              tabRef={[Function]}
            >
              <li
                aria-controls="react-tabs-1"
                aria-disabled="false"
                aria-selected="true"
                className="react-tabs__tab react-tabs__tab--selected"
                id="react-tabs-0"
                role="tab"
                tabIndex="0"
              >
                Title 1
              </li>
            </Tab>
            <Tab
              className="react-tabs__tab"
              disabledClassName="react-tabs__tab--disabled"
              focus={false}
              id="react-tabs-2"
              key=".1"
              panelId="react-tabs-3"
              selected={false}
              selectedClassName="react-tabs__tab--selected"
              tabRef={[Function]}
            >
              <li
                aria-controls="react-tabs-3"
                aria-disabled="false"
                aria-selected="false"
                className="react-tabs__tab"
                id="react-tabs-2"
                role="tab"
                tabIndex={null}
              >
                Title 2
              </li>
            </Tab>
          </ul>
        </TabList>
        <TabPanel
          className="react-tabs__tab-panel"
          forceRender={false}
          id="react-tabs-1"
          key=".1"
          selected={true}
          selectedClassName="react-tabs__tab-panel--selected"
          tabId="react-tabs-0"
        >
          <div
            aria-labelledby="react-tabs-0"
            className="react-tabs__tab-panel react-tabs__tab-panel--selected"
            id="react-tabs-1"
            role="tabpanel"
          >
            <button>
              Edut1
            </button>
          </div>
        </TabPanel>
        <TabPanel
          className="react-tabs__tab-panel"
          forceRender={false}
          id="react-tabs-3"
          key=".2"
          selected={false}
          selectedClassName="react-tabs__tab-panel--selected"
          tabId="react-tabs-2"
        >
          <div
            aria-labelledby="react-tabs-2"
            className="react-tabs__tab-panel"
            id="react-tabs-3"
            role="tabpanel"
          />
        </TabPanel>
      </div>
    </UncontrolledTabs>
  </Tabs>
</EditComponent>
`;
